src/libraries/Microsoft.Extensions.Caching.Abstractions/src/MemoryCacheExtensions.cs

```diff
@@ -99,13 +103,11 @@ public static TItem GetOrCreate<TItem>(this IMemoryCache cache, object key, Func
         {
             if (!cache.TryGetValue(key, out object result))
             {
-                ICacheEntry entry = cache.CreateEntry(key);
-                result = factory(entry);
-                entry.SetValue(result);
-                // need to manually call dispose instead of having a using
-                // in case the factory passed in throws, in which case we
-                // do not want to add the entry to the cache
-                entry.Dispose();
+                using (ICacheEntry entry = cache.CreateEntry(key))
+                {
+                    result = factory(entry);
+                    entry.SetValue(result);
```

Comment:
Why are some of these cases using `entry.Value = value;` while others are using `entry.SetValue`?  Seems like the latter is just a wrapper for the former that also returns the entry, but the return value is ignored here.

Reply:
You are correct that's all `entry.SetValue` provides over `entry.Value`.

Fixed.